IBM TTS

bobbyshaw opened the pull request.

bobbyshaw committed:
Add new items status for failed conversion. Then use this if we get an incorrect response or an exception from the API.

bobbyshaw committed:
Flash error message rather than it being persistant.

bobbyshaw committed:
Use IBM Watson to TTS.

bobbyshaw committed:
Autoindent.

tgerulaitis commented on app/Console/Commands/ConvertToSpeech.php:
```diff
@@ -57,29 +59,65 @@ public function fire()
+                    try {
+                        $response = $this->getHttpClient()->post("v1/synthesize", array(
+                            "headers" => array(
+                                'Authorization' => 'Basic ' . base64_encode(env('WATSON_TTS_USERNAME') . ':' . env('WATSON_TTS_PASSWORD')),
```
Probably better to use the `auth` option and let Guzzle encode the authentication header for you (http://guzzle.readthedocs.org/en/latest/request-options.html#auth)

tgerulaitis commented on app/Console/Commands/ConvertToSpeech.php:
```diff
@@ -57,29 +59,65 @@ public function fire()
+                        if ($returnCode !== 0) {
+                            $this->error('Error converting .ogg to .mp3');
+                            $item->status = Items::STATUS_CONVERSION_FAILED;
+                            $item->save();
```
We should probably `break;` here, so we don't output a successful message (and don't delete the `ogg` file in case we just want to reconvert it?) if the conversion to `mp3` failed.

tgerulaitis commented on app/Console/Commands/ConvertToSpeech.php:
```diff
@@ -57,29 +59,65 @@ public function fire()
-                    } else {
-                        printf("Failed!\n");
+
+                        $this->info(sprintf("Done (%s).", $filename));
```
The file that's in `$filename` here no longer exists as we've converted it to `mp3` and deleted the original.

tgerulaitis commented:
:+1:

bobbyshaw committed:
Update based on Tomas' feedback.

- Use Guzzle auth
- Ensure we only error when we intend to and when we do, continue on to the next item in the for loop.
-

bobbyshaw commented:
Thanks a lot Tomas!
